src/frontend/src/common/components/library/data_subview/components/UsherTreeFlow/components/UsherPreparingModal/index.tsx

```diff
@@ -26,7 +29,9 @@ const UsherPreparingModal = ({
     if (!fastaUrl) return;
 
     const link = document.createElement("a");
-    link.href = `${USHER_URL + fastaUrl}`;
+    link.href = `${USHER_URL}${encodeURIComponent(
```

Comment:
The process of generating the redirect URL is getting complicated enough it feels like it should be extracted. Maybe pulled to a standalone Usher "API" file, maybe just a function that's defined in this file, but it seems complicated enough to me that it's better as its own process.